### PR TITLE
Fix postgresql set transaction isolation

### LINKED ISSUE #22493: Can't set transaction isolation in postgresql
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
master

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
proxy

### Expected behavior
Can't execute `set transaction isolation level serializable;` in proxy.

```sql
-- set serializable isolation
sharding_db=>  set transaction isolation level serializable;
SET

-- but not success
sharding_db=> show transaction_isolation;
 transaction_isolation
-----------------------
 read committed
(1 row)
```

### Actual behavior
Can execute  `set transaction isolation level serializable;` success in proxy and take effect.

### Reason analyze (If you can)
`PostgreSQLTCLStatementSQLVisitor` is not parsing this sql.

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
Refer to this class `MySQLTCLStatementSQLVisitor`, add the similar logic for postgresql.

### PR DESCRIPTION
**Pull request by FlyingZC**
Fixes #22493
Changes proposed in this pull request:
  - Fix postgresql set transaction isolation

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Review by strongduanmu — CHANGES_REQUESTED** [review]
Please submit an issue to describe this bug.

**Comment by FlyingZC** [comment]
> 

OK,I have submitted the issue.

File: sql-parser/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLTCLStatementSQLVisitor.java [hunk 1]
```diff
@@ -68,7 +71,40 @@ public PostgreSQLTCLStatementSQLVisitor(final Properties props) {
...
+        TransactionAccessType accessType = null;
+        if (null != modeItemContext.ONLY()) {
+            accessType = TransactionAccessType.READ_ONLY;
+        } else if (null != modeItemContext.WRITE()) {
+            accessType = TransactionAccessType.READ_WRITE;
+        }
+        statement.setAccessMode(accessType);
+    }
+    
+    private void setTransactionIsolationLevel(final PostgreSQLSetTransactionStatement statement, final TransactionModeItemContext modeItemContext) {
```
Comment (strongduanmu) [inline sql-parser/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLTCLStatementSQLVisitor.java:94]: Can we refactor this method to getTransactionIsolationLevel and then use guard clause?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: sql-parser/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLTCLStatementSQLVisitor.java (hunk 1 above)
Comment (FlyingZC) [reply sql-parser/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLTCLStatementSQLVisitor.java]: OK.

**Comment by FlyingZC** [comment]
image

**Review by strongduanmu — APPROVED** [review]

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**